classes/Link.php

```diff
@@ -499,29 +499,31 @@ public function getLanguageLink($id_lang, Context $context = null)
 		if (!empty($context->controller->php_self))
 			$controller = $context->controller->php_self;
 
+		$id_shop = (int)$context->shop->id;
```

Comment:
@julienbourdeau the point is HERE ... The method accept a Context, but does not fully honor it ...